src/phoenix/server/api/mutations/api_key_mutations.py

```diff
@@ -79,6 +91,22 @@ async def create_system_api_key(
             query=Query(),
         )
 
+    @strawberry.mutation(permission_classes=[HasSecret, IsAuthenticated])  # type: ignore
+    async def delete_system_api_key(
+        self, info: Info[Context, None], input: DeleteApiKeyInput
+    ) -> DeleteSystemApiKeyMutationPayload:
+        api_key_id = from_global_id_with_expected_type(
+            input.id, expected_type=SystemApiKey.__name__
+        )
+        async with info.context.db() as session:
+            api_key = await session.get(models.APIKey, api_key_id)
+            if api_key is None:
+                raise ValueError("API Key not found")
```

Comment:
We need to use `phoenix.server.api.errors.NotFound` here, otherwise, this error message will not be returned to the caller.

Reply:
You may need to merge latest to get [this change](https://github.com/Arize-ai/phoenix/pull/4339).